src/test/java/org/broadinstitute/hellbender/tools/copynumber/CreateBinningIntervalsIntegrationTest.java

```diff
@@ -0,0 +1,221 @@
+                "-" + CreateBinningIntervals.PADDING_SHORT_NAME, Integer.toString(padding),
+                "-O", outputFile.getAbsolutePath()
+        };
+        runCommandLine(args);
+
+        // read the contents of the output file and print them out
+        // printOutputFile(outputFile);
+
+        // read the bins from the output file
+        IntervalList binsFromFile = IntervalList.fromFile(outputFile);
+
+        // we expect the following result
+        IntervalList binsExpected = new IntervalList(binsFromFile.getHeader().getSequenceDictionary());
+        String contig = binsFromFile.iterator().next().getContig();
+        binsExpected.add(new Interval(contig, 1, 10000000));
+        binsExpected.add(new Interval(contig, 10000001, 20000000));
+        binsExpected.add(new Interval(contig, 20000001, 30000000));
+        binsExpected.add(new Interval(contig, 30000001, 40000000));
+        binsExpected.add(new Interval(contig, 40000001, 50000000));
+        binsExpected.add(new Interval(contig, 50000001, 60000000));
+        binsExpected.add(new Interval(contig, 60000001, 63025520));
+
+        // compare the file contents to the expected result
+        Assert.assertEquals(binsFromFile, binsExpected);
+    }
+
+
+    // Test for reading intervals from a file
+    @Test
+    public void singleFileTest5() {
+        final int widthOfBins = 10000;
+        final int padding = 5000;
+        final File outputFile = createTempFile("GATK-create-binning-intervals-test", ".tmp");
+        final String[] args = {
+                "-L", "src/test/resources/org/broadinstitute/hellbender/tools/copynumber/interval-preprocessor-intervals.interval_list",
+                "-R", REFERENCE.getAbsolutePath(),
+                "-" + CreateBinningIntervals.WIDTH_OF_BINS_SHORT_NAME, Integer.toString(widthOfBins),
+                "-" + CreateBinningIntervals.PADDING_SHORT_NAME, Integer.toString(padding),
+                "-O", outputFile.getAbsolutePath()
+        };
+        runCommandLine(args);
+
+        // read the contents of the output file and print them out
+        // printOutputFile(outputFile);
+
+        // read the bins from the output file
+        IntervalList binsFromFile = IntervalList.fromFile(outputFile);
+
+        // we expect the following result
+        IntervalList binsExpected = new IntervalList(binsFromFile.getHeader().getSequenceDictionary());
+        binsExpected.add(new Interval("20", 1, 10000));
+        binsExpected.add(new Interval("20", 10001, 16000));
+        binsExpected.add(new Interval("21", 1, 5100));
+        binsExpected.add(new Interval("21", 15000, 24999));
+        binsExpected.add(new Interval("21", 25000, 27000));
+
+        // compare the file contents to the expected result
+        Assert.assertEquals(binsFromFile, binsExpected);
+
+    }
+
+    private static void printOutputFile(final File outputFile) {
```

Comment:
This method is not used anymore

Reply:
Please use the public final variable `packageRootTestDir` from the `BaseTest` class (a superclass of `CommandLineProgramTest`) in place of `src/test/resources/org/broadinstitute/hellbender/`. This would reduce the risk that a possible reorganization of the codebase in the future would break your test. Also, it is a good idea to extract a variable `private static final File TEST_INTERVAL_LIST_FILE = new File(packageRootTestDir + "tools/copynumber/interval-preprocessor-intervals.interval_list")` and use that instead. The idea is to extract as much _data_ as possible out of the test methods, either by means of using data providers or static final variables defined in the preamble of the test class.

Reply:
The constructor `new File(root, relative)` is preferable to `new File(root + relative)`.

Reply:
Ok, thanks for these comments. I kept that method because I wanted to use it for testing in the second round of thee code review. I deleted it now.